### ISSUE TITLE
Bootstrap fails to find llvm-config

### ISSUE THREAD

**Issue by ZuseZ4** [rust-lang/rust#137158]
### Summary

Thanks for updating the llvm version check logic, but it fails for me

### Command used

```sh
./x.py build --stage 1 library
```

### Expected behaviour

It worked a few times.
### Actual behaviour
Now it starts failing the unwrap here:
`let llvm_config = builder.llvm_config(builder.config.build).unwrap();`

```sh
➜  rust-middle git:(master) ✗ ./x.py build --stage 1 library            
Building bootstrap
    Finished `dev` profile [unoptimized] target(s) in 0.04s
WARNING: The `change-id` is missing in the `config.toml`. This means that you will not be able to track the major changes made to the bootstrap configurations.
NOTE: to silence this warning, add `change-id = 136941` at the top of `config.toml`
[src/bootstrap/src/core/builder/mod.rs:1439:9] "llvm_config" = "llvm_config"
[src/bootstrap/src/core/builder/mod.rs:1440:9] self.config.llvm_enabled(target) = true
[src/bootstrap/src/core/builder/mod.rs:1441:9] self.kind != Kind::Check = true
[src/bootstrap/src/core/builder/mod.rs:1442:9] !self.config.dry_run() = false

thread 'main' panicked at src/bootstrap/src/core/build_steps/compile.rs:1096:69:
called `Option::unwrap()` on a `None` value
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
Build completed unsuccessfully in 0:00:00
```
If I remove the dry-run check it also fails:
```
thread 'main' panicked at src/bootstrap/src/core/build_steps/llvm.rs:630:45:
Failed to parse LLVM version
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
Build completed unsuccessfully in 0:00:03
```

### Bootstrap configuration (config.toml)
```toml
./configure --enable-llvm-link-shared --enable-llvm-enzyme --release-channel=nightly --enable-llvm-assertions --enable-clang --enable-lld --enable-option-checking --enable-ninja  --enable-docs
```

### HEAD

d1fb81e8dd5354ddf7cb334d5a234cab7f64b3bb

### Additional context
I tried in my pr, (with my own changes to it removed) as well as master d1fb81e8dd5354ddf7cb334d5a234cab7f64b3bb
The strange thing it didn't fail for a few times. 

Also,
```sh
➜  rust-middle git:(master) ✗ find build/x86_64-unknown-linux-gnu -name llvm-config
build/x86_64-unknown-linux-gnu/llvm/bin/llvm-config
build/x86_64-unknown-linux-gnu/llvm/build/tools/llvm-config
build/x86_64-unknown-linux-gnu/llvm/build/bin/llvm-config
```

Also, rm -rf on master did not help.

<details><summary>Build Log</summary>
<p>

```txt
<log>
```

</p>
</details>

r? @onur-ozkan

**Comment by ZuseZ4** [comment]
Oh, issues have no reviewers, makes sense.
Let's see if this works.

@rustbot assign  @onur-ozkan

**Comment by ZuseZ4** [comment]
Ok, I think I figured out the reason. When I want to be 100% sure that my submodule changes are compiled, then I do
`rm -rf build/x86_andsoon/enzyme`. I didn't delete `build/x86_something/llvm`, but I assume that confused something?

I'm not sure if I'm just not supposed to do that, and just trust that src/tools/enzyme submodule changes are considered?

**Comment by ZuseZ4** [comment]
Ok, so after some more experimenting, it turns out that in check builds we don't need to link enzyme sind it's a check build, so instead of unwrapping we just do nothing.
Does that sound like the right solution, or should we maybe even skip the whole step in check builds? I'm not sure if that's possible or expected.

**Comment by onur-ozkan** [comment]
I think unwrapping should stay as it is because we don't want to silently skip enzyme when there is an unexpected error. Skipping enzyme for check builds sounds like the right approach.

**Comment by ZuseZ4** [comment]
Good point, I adjusted it, thanks.
